packages/runtime-handler/src/dev-runtime/internal/functionRunner.ts

```diff
@@ -102,7 +102,11 @@ process.on(
           `Could not find a "handler" function in file ${functionPath}`
         );
       }
-      handler(context, event, callback);
+      if (handler.constructor.name === 'AsyncFunction') {
+        await handler(context, event, callback);
```

Comment:
Is this causing any issues in terms of functionality parity with production Functions? I'm worried whether this will await things that real Functions would ultimately cancel

Reply:
I don't think so. Once you call the `callback` (with either a success or error) messages are passed up to the parent and the process ended. This just exists to catch errors thrown by the function and handle them by also passing to the parent and closing the process.

One thing that we don't have in the runtime-handler here is cancellation of functions after 10 seconds, but that's true for non-async functions too.

I guess one question is, what happens in production Functions if you have an `async` function that throws an error for some reason?

One other thing, this actually allows for the dev runtime-handler to render the descriptive error page that you implemented. So in terms of parity with prod functions, I'm not sure where that lies!